### PR TITLE
Fix for issue where providers were sending user to the wrong film

### PR DESCRIPTION
**Pull request by rjs68**
-Provides a fix for the issue where the user was being directed to the wrong movie
-Ensures the movie name returned by utelly is equal to the one we provided

### CODE REVIEW

File: pickee_api/gateway.py [hunk 1]
```diff
@@ -37,7 +37,11 @@ def get_available_providers(request):
 
         response = {}
         if data.get('results'):
-            relevant_result = data.get('results')[0]
+            result_index = 0
+            relevant_result = data.get('results')[result_index]
+            while relevant_result.get('name') != movie_name:
```
Comment (BelfDev) [inline pickee_api/gateway.py]: Are comparing here the TMDB movie name with Utelly's movie name? It so maybe we should use some logic that verifies whether it is a close match.

Comment (rjs68) [reply pickee_api/gateway.py]: This is a good point. I'll look into it!

**Comment by Arden488** [comment]
Rhys, I think the most efficient way would be to filter if there is an exact match in the results (because we know the exact movie title). If there is a match - send it, otherwise - send an empty response.
For example, check the "Joker" query. It returns 4 results and there is a "Joker" movie in them, but it is on the 3rd index. We can efficiently discard all other results and leave only the 3rd one.

**Comment by rjs68** [comment]
> Rhys, I think the most efficient way would be to filter if there is an exact match in the results (because we know the exact movie title). If there is a match - send it, otherwise - send an empty response.
> For example, check the "Joker" query. It returns 4 results and there is a "Joker" movie in them, but it is on the 3rd index. We can efficiently discard all other results and leave only the 3rd one.

@Arden488 I have just had a look at the filter methods in python but I can't get my head around them!

**Pull request merged**